cloudbaseinit/metadata/services/vmwareguestinfoservice.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2019 ruilopes.com
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import base64
+import gzip
+import io
+import json
+import os
+
+from cloudbaseinit import conf as cloudbaseinit_conf
+from cloudbaseinit import exception
+from cloudbaseinit.metadata.services import base
+from cloudbaseinit.osutils import factory as osutils_factory
+
+CONF = cloudbaseinit_conf.CONF
+
+class VMwareGuestInfoService(base.BaseMetadataService):
+    """
+    This uses the VMware Guest Info interface to obtain the cloud init data
+    from the VM extraconfig guestinfo properties using the VMware Tools
+    rpctool cli application.
+
+    You can use the following guestinfo properties:
+
+    +--------------------+------------------------------------------------------+
+    | property           | description                                          |
+    +--------------------+------------------------------------------------------+
+    | guestinfo.metadata | A JSON document containing the cloud-init metadata.  |
+    | guestinfo.userdata | A YAML document containing the cloud-init userdata,  |
+    |                    | or a MIME multipart message with the YAML document   |
+    |                    | containing the cloud-init user data and custom       |
+    |                    | scripts as described at doc/source/userdata.rst      |
+    +--------------------+------------------------------------------------------+
+
+    Each property value should be gzip compressed and must be base64 encoded.
```

Comment:
Just to clarify - the _should_ here is intentional, correct? In that https://github.com/cloudbase/cloudbase-init/pull/20/files#diff-79a52948a1c2159908ab0bec485abdeaR130-R132 suggest that gzipping is not required.

Reply:
Yes, correct.